File: benchmark/nodejs/matrix_transpose.js
const size = 300;
const m = [];
const t = [];
for (let i = 0; i < size; i++) {
    const row_m = [];
    const row_t = [];
    for (let j = 0; j < size; j++) {
        row_m.push(i + j);
        row_t.push(0);
    }
    m.push(row_m);
    t.push(row_t);
}

const start = Date.now();
for (let i = 0; i < size; i++) {
    for (let j = 0; j < size; j++) {
        t[j][i] = m[i][j];
    }
}
const end = Date.now();
console.log(`T[0][0]: ${t[0][0]}`);
console.log(`T[299][299]: ${t[299][299]}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/ruby/matrix_transpose.rb
size = 300
m = []
t = []
for i in 0...size
    row_m = []
    row_t = []
    for j in 0...size
        row_m.push(i + j)
        row_t.push(0)
    end
    m.push(row_m)
    t.push(row_t)
end

start = Time.now.to_f * 1000
for i in 0...size
    for j in 0...size
        t[j][i] = m[i][j]
    end
end
finish = Time.now.to_f * 1000
puts "T[0][0]: #{t[0][0]}"
puts "T[299][299]: #{t[299][299]}"
puts "Time: #{(finish - start).round}ms"

